src/happy/oauth2_capture_redirect.clj

```diff
@@ -14,19 +14,67 @@
     (catch Exception _
       nil)))
 
-(defn wait-for-redirect [{:as config :keys [redirect_uri redirect_uris]}]
+(defn redirect-uri->port
+  "Opening a browser and redirecting to a local server
+  only makes sense for localhost redirect-uris.
+
+  Known limitation: Google does not support no ipv6 [::1] with random ports"
+  [uri]
+  (let [[match protocol host _ port path] (re-find #"^(http://)(localhost|127.0.0.1)(:(\d+))?(/.*)?$" uri)
+        port (maybe-parse-int port)]
+    (if (not match)
+      (throw (RuntimeException. "URI doesn't match pattern"))
+      (or port
+          0)))) ;; Jetty's magic value 0 = random port
+
+(defn add-port
+  [uri random-port]
+  (let [[match protocol host _ port path] (re-find #"^(http://)(localhost|127.0.0.1)(:(\d+))?(/.*)?$" uri)]
```

Comment:
I considered extracting the pattern to avoid redundancy, but decided against it in favor of easier readability.

I protected the code against mistakes by adding some unit tests.